x-pack/test/ui_capabilities/common/services/ui_capabilities.ts

```diff
@@ -100,6 +96,61 @@ export class UICapabilitiesService {
       );
     }
   }
+
+  public async getWithNavLinks(
+    credentials: BasicCredentials | null,
+    navLinks: Record<string, boolean> = {},
+    spaceId?: string
+  ): Promise<GetUICapabilitiesResult> {
+    // Get the base capabilities
+    const { success, failureReason, value: baseCapabilities } = await this.get(
+      credentials,
+      spaceId
+    );
+    if (!success || !baseCapabilities) {
+      return { success, failureReason };
+    }
+
+    const spaceUrlPrefix = spaceId ? `/s/${spaceId}` : '';
+    this.log.debug(`requesting ${spaceUrlPrefix}/app/kibana to parse the uiCapabilities`);
+    const requestOptions = credentials
+      ? {
+          auth: credentials,
+        }
+      : {};
+    const response = await this.axios.post(
+      `${spaceUrlPrefix}/api/capabilities`,
+      { capabilities: { ...baseCapabilities, navLinks } },
+      requestOptions
+    );
+
+    if (response.status === 302 && response.headers.location === '/') {
+      return {
+        success: false,
+        failureReason: GetUICapabilitiesFailureReason.RedirectedToRoot,
+      };
+    }
+
+    if (response.status === 404) {
+      return {
+        success: false,
+        failureReason: GetUICapabilitiesFailureReason.NotFound,
+      };
+    }
+
+    if (response.status !== 200) {
+      throw new Error(
```

Comment:
Can you explain the purpose of this throwing instead of having a failure response?

Reply:
If a service throws an error during a lifecycle event, it automatically gets handled by the FatalErrorService which will show a dialog. I want this to fail hard by blocking progress on anything else and showing an error.